tests/test_project.py

```diff
@@ -281,13 +297,30 @@ def test_find_jobs(self):
         assert 0 == len(list(self.project.find_jobs({"sp.a": 5})))
         assert 1 == len(list(self.project.find_jobs(doc_filter={"b": 0})))
         assert 0 == len(list(self.project.find_jobs(doc_filter={"b": 5})))
+        assert 1 == len(list(self.project.find_jobs(None, {"b": 0})))
+        assert 0 == len(list(self.project.find_jobs(None, {"b": 5})))
         assert 1 == len(list(self.project.find_jobs({"doc.b": 0})))
         assert 0 == len(list(self.project.find_jobs({"doc.b": 5})))
         assert 1 == len(list(self.project.find_jobs({"a": 0, "doc.b": 0})))
+        assert 1 == len(list(self.project.find_jobs({"a": 0}, {"b": 0})))
+        assert 1 == len(
+            list(self.project.find_jobs(filter={"a": 0}, doc_filter={"b": 0}))
+        )
+        assert 1 == len(list(self.project.find_jobs({"a": 0}, doc_filter={"b": 0})))
         assert 1 == len(list(self.project.find_jobs({"sp.a": 0, "doc.b": 0})))
+        assert 0 == len(list(self.project.find_jobs({"a": 0, "doc.b": 5})))
+        assert 0 == len(list(self.project.find_jobs({"a": 0}, {"b": 5})))
+        assert 0 == len(
+            list(self.project.find_jobs(filter={"a": 0}, doc_filter={"b": 5}))
+        )
+        assert 0 == len(list(self.project.find_jobs({"a": 0}, doc_filter={"b": 5})))
         assert 0 == len(list(self.project.find_jobs({"sp.a": 0, "doc.b": 5})))
         assert 0 == len(list(self.project.find_jobs({"sp.a": 5, "doc.b": 0})))
         assert 0 == len(list(self.project.find_jobs({"sp.a": 5, "doc.b": 5})))
+        with pytest.raises(ValueError):
+            self.project.find_jobs({"a": 0}, {"b": 5}, "invalid positional arg")
+        with pytest.raises(ValueError):
+            self.project.find_jobs({"a": 0}, invalid_kwarg="invalid argument")
```

Comment:
Please add the tests for too many arguments including with key-word arguments (`(..., ..., doc_filter=...)`).

Reply:
If I understand your suggestion correctly, this is prevented by Python's function calling syntax. Did I miss something?

```python
>>> def func(a=1, b=2):
...   print(a, b)
...
>>> func(4, 5, b=7)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: func() got multiple values for argument 'b'
```

Reply:
But `find_jobs({"a": 0}, {"b: 0}", doc_filter={"c": 0})` would be technically valid, but should raise an error.

Reply:
Resolved in fc539938c1efaa693f17fb61d17795dddec3c8eb.

Reply:
Resolved in 181dca732ab876e4bd430b40c78c07076a6f5637.